packages/dask-task-models-library/src/dask_task_models_library/container_tasks/events.py

```diff
@@ -27,19 +30,42 @@ def topic_name() -> str:
         return "task_progress"
 
     @classmethod
-    def from_dask_worker(cls, progress: float) -> "TaskProgressEvent":
-        return cls(job_id=get_worker().get_current_task(), progress=progress)
+    def from_dask_worker(
+        cls, progress: float, *, task_owner: TaskOwner
+    ) -> "TaskProgressEvent":
+        worker = get_worker()
+        job_id = worker.get_current_task()
+
+        return cls(
+            job_id=job_id,
+            progress=progress,
+            task_owner=task_owner,
+        )
 
     class Config(BaseTaskEvent.Config):
-        schema_extra = {
+        schema_extra: ClassVar[dict[str, Any]] = {
             "examples": [
                 {
                     "job_id": "simcore/services/comp/sleeper:1.1.0:projectid_ec7e595a-63ee-46a1-a04a-901b11b649f8:nodeid_39467d89-b659-4914-9359-c40b1b6d1d6d:uuid_5ee5c655-450d-4711-a3ec-32ffe16bc580",
                     "progress": 0,
+                    "task_owner": {
+                        "user_id": 32,
+                        "project_id": "ec7e595a-63ee-46a1-a04a-901b11b649f8",
+                        "node_id": "39467d89-b659-4914-9359-c40b1b6d1d6d",
+                        "parent_project_id": None,
+                        "parent_node_id": None,
+                    },
```

Comment:
well, then where do you stop? :rofl: 
But seriously I think with this system we already have the whole hierarchy since we can follow the parents until there is none and that would be the ancestor no?

Reply:
Yep, good point.